### PR TITLE
[design doc]: support Analyze SQL in SQLFlow

### PR DESCRIPTION
**Pull request by Yancey0623**
A part task of #707

### CODE REVIEW

File: doc/analyzer.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- `force` and `summary` is the visualized method.
+- `TreeExplainer` is the [explain type](https://github.com/slundberg/shap#sample-notebooks).
+
+The **Analyze SQL** would display the visualization image on Jupyter like:
+<img src="https://raw.githubusercontent.com/slundberg/shap/master/docs/artwork/boston_dataset.png">
+
+## Implement Details
+
+- Enhance the SQLFlow parser to support the `ANALYZE` keyword.
+- implement the `codgen_shap.go` to generate a SHAP Python program to save the visualization image as HTML format.
```
Comment (tonyyang-svail) [inline doc/analyzer.md]: Some suggestions for elaborating the details:

The `codgen_shap.go` generates a SHAP Python program. The SHAP Python program prints the resulting image in HTML format to stdout. The stdout will be captured by the Go program using [`CombinedOutput`](https://golang.org/pkg/os/exec/#Cmd.CombinedOutput). The Go program then sends the HTML text to the client as a single message.

**Comment by Echo9573** [comment]
Overall, the interpretative needs of current analysts have been largely met. 
A little suggestion about this design document: Since TreeExplainer shap only supports tree models, it is not recommended to use "my_dnn_model" instead of tree_model or others in Analyze SQL Syntax example。

**Comment by Yancey0623** [comment]
> Overall, the interpretative needs of current analysts have been largely met.
A little suggestion about this design document: Since shap only supports tree models, it is not recommended to use "my_dnn_model" instead of tree_model or others in Analyze SQL Syntax example。

Hi @Echo9573 , thanks for your suggestion, I will fix the typo `my_dnn_model` to `my_model`. And maybe shap can support the Tree Model using `TreeExplainer` and also the deep learning model using `DeepExplainer` or `GradientExplainer `? Ref: https://github.com/slundberg/shap.

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

File: doc/analyzer_design.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+INTO my_model;
+```
+
+Analyze SQL:
+
+``` sql
+SELECT * FROM train_table
+ANALYZE my_model
+WITH
+  plots = force
```
Comment (typhoonzero) [inline doc/analyzer_design.md:31]: Maybe add some comments about how to select the axis to plot like `shap.force_plot(explainer.expected_value, shap_values[0,:], X.iloc[0,:])`

Comment (Yancey0623) [reply doc/analyzer_design.md:31]: Will update this section in the next PR.

**Pull request merged**